devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1035,6 +1163,366 @@ def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejec
+				} else {
+					log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule set failed for codeID := $codeID"
+					result << createEvent(name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_DAILY_REPEATING failed", descriptionText: "Recurring schedule set failed for \"$codeName\"", isStateChange: true, displayed: false)
+				}
+			} else {
+				log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule found for codeID := $codeID while scanning"
+				def endHour = cmd.startHour + cmd.durationHour
+				def endMin = cmd.startMinute + cmd.durationMinute
+				if (endMin >= 60) {
+					endMin = endMin % 60
+					endHour = endHour + 1
+				}
+				
+				state["setDailyRepeatingStartTime$codeID"] = sprintf("%02d:%02d", cmd.startHour, cmd.startMinute)
+				state["setDailyRepeatingEndTime$codeID"] = sprintf("%02d:%02d", endHour, endMin)
+				state["setDailyRepeatingBitMask$codeID"] = cmd.weekDayBitmask
+				result << scheduleSetEvent(codeID, SCHEDULE_TYPE_DAILY_REPEATING)
+				def map = [name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_DAILY_REPEATING $changeType", descriptionText: "${getStatusForDescription(changeType)} recurring schedule for \"$codeName\"", isStateChange: true, displayed: true]
+				if (changeType == "changed") {
+					map.displayed = false
+				}
+				result << createEvent(map)
+				
+				// schedule scanning for this codeID is complete
+				log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule scanning complete for codeID := $codeID"
+				checkAndScanNextSchedule(codeID)
+			}
+			clearDailyRepeatingScheduleDataFromState(codeID)
+		}
+
+		if (result) {
+			result = result.flatten()
+			return result
+		}
+		return null
+	}
+	return null
+}
+
+/**
+ * Responsible for parsing ScheduleEntryLockWeekDayReport command
+ *
+ * @param cmd: The ScheduleEntryLockWeekDayReport command to be parsed
+ *
+ * @return The event(s) to be sent out
+ *
+ */
+def zwaveEvent(physicalgraph.zwave.commands.scheduleentrylockv3.ScheduleEntryLockWeekDayReport cmd) {
+	log.trace "[DTH] Executing 'zwaveEvent(physicalgraph.zwave.commands.scheduleentrylockv3.ScheduleEntryLockWeekDayReport cmd)' with cmd = $cmd"
+	if (device.currentValue("numberOfSlotsWeekDay")) {
+		def result = []
+		def codeID = cmd.userIdentifier
+		def lockCodes = loadLockCodes()
+		def codeName = getCodeName(lockCodes, codeID)
+		
+		def daysOfWeek = state["setWeekDayWeekDays$codeID"]
+		if (daysOfWeek && daysOfWeek.size() > 0) {
+			int dayOfWeek = cmd.dayOfWeek
+			if (dayOfWeek in daysOfWeek) {
+				def startTime = getWeekDayHoursAndMinutes(state["setWeekDayStartTime$codeID"])
+				def startHour = startTime.hour
+				def startMin = startTime.minute
+				
+				def endTime = getWeekDayHoursAndMinutes(state["setWeekDayEndTime$codeID"])
+				def endHour = endTime.hour
+				def endMin = endTime.minute
+				
+				if (cmd.startHour == startHour && cmd.startMinute == startMin && cmd.stopHour == endHour && cmd.stopMinute == endMin) {
```

Comment:
As a general comment, can this be commented a bit more. It's not entirely clear at a glance what is happening in this code.